controllers/numaresourcesoperator_controller.go

```diff
@@ -179,21 +179,22 @@ func (r *NUMAResourcesOperatorReconciler) Reconcile(ctx context.Context, req ctr
 }
 
 // updateStatusConditionsIfNeeded returns true if conditions were updated.
-func updateStatusConditionsIfNeeded(instance *nropv1.NUMAResourcesOperator, condition string, reason string, message string) {
-	if condition == "" { // backward (=legacy) compatibility
+func updateStatusConditionsIfNeeded(instance *nropv1.NUMAResourcesOperator, cond conditionInfo) {
+	if cond.Type == "" { // backward (=legacy) compatibility
 		return
 	}
-	klog.InfoS("updateStatus", "condition", condition, "reason", reason, "message", message)
-	conditions, ok := status.UpdateConditions(instance.Status.Conditions, condition, reason, message)
+	klog.InfoS("updateStatus", "condition", cond.Type, "reason", cond.Reason, "message", cond.Message)
+	conditions, ok := status.UpdateConditions(instance.Status.Conditions, cond.Type, cond.Reason, cond.Message)
 	if ok {
 		instance.Status.Conditions = conditions
 	}
 }
 
 func (r *NUMAResourcesOperatorReconciler) degradeStatus(ctx context.Context, instance *nropv1.NUMAResourcesOperator, reason string, stErr error) (ctrl.Result, error) {
-	message := messageFromError(stErr)
+	info := degradedConditionInfoFromError(stErr)
+	info.Reason = reason
```

Comment:
if `reason` is empty it will override `info.Reason` (keep it as "InternalError"?) if we set it above `Reason: reasonFromError(err)`

Reply:
indeed, good catch. I think I fixed in the later commits by rearranging the flow.